### PR TITLE
Try at npm init

### PR DESCRIPTION
**Pull request by msecret**
Would like feedback on structure, did I put things in right place? Is it better to prompt the user in case they don't want front end deps at all?

I think keeping all the scaffolding folders in the base of the repo will get confusing, although might be fine if all this repo has is generators.

### CODE REVIEW

File: npm/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }]).then((answers) => {
+      this.frontendDeps = answers.frontendDeps;
+    });
+  },
+
+  configuring: function() {
+    if (this.frontendDeps) {
+      this.log('Configuring npm now...');
+      this.composeWith(require.resolve('generator-npm-init/app'), {
+        repo: 'https://github.com/18F/<%= destFolderName %>',
```
Comment (cmc333333) [inline npm/index.js]: This bit isn't quite working; I'm getting the string `https://github.com/18F/<%= destFolderName %>`. Maybe we need to use `this.destinationPath()` and straight string concatenation?

File: package.json [hunk 2]
```diff
@@ -10,10 +10,15 @@
   "bugs": {
     "url": "https://github.com/18F/18f-cli/issues"
   },
-  "files": ["app", "license", "readme"],
+  "files": [
+    "app",
+    "license",
+    "readme"
+  ],
```
Comment (cmc333333) [inline package.json:18]: Does this need to include the `npm` folder?

**Review by cmc333333 — COMMENTED** [review]
100 points for reusing the npm-init generator. Left a few comments on potential bugs, but I think this is the right direction.

Re: prompts -- I think it makes sense to ask here; we build enough APIs and other non-frontend apps that it's worth giving the opt out. Would it make sense to store this as a `this.config`? I'm not sure if anything else would key off it.

Re directory layout: :shrug: doesn't make a huge difference to me.

File: npm/index.js (hunk 1 above)
Comment (msecret) [reply npm/index.js]: yea probably

File: package.json (hunk 2 above)
Comment (msecret) [reply package.json:18]: hmm maybe, although it did work. This might be another reason to nest all of them under a `generators` folder.

Comment (cmc333333) [reply package.json:18]: Maybe it needs to be included once we ship to npm? :shrug: not my forte

**Comment by cmc333333** [comment]
One other weird bit: the `package.json` it generated for be includes this key-value:

```javascript
"_": []
```

Are you seeing something similar?

**Comment by msecret** [comment]
I'm getting that too @cmc333333, it might be setting up as a default for dependencies? I'm not really sure.

**Comment by cmc333333** [comment]
Rockin. We can figure out the blank thing later.

**Comment by rogeruiz** [comment]
Late to the party, but I wonder if this would be helpful at all? https://github.com/ahmadnassri/npm-package-generator

**Pull request merged**